File: scope.js
let x = 10; // Global scope

function outer() {
  let x = 20; // Enclosing scope
  function inner() {
    let x = 30; // Local scope
    console.log('Inner x:', x);
  }
  inner();
  console.log('Outer x:', x);
}

outer();
console.log('Global x:', x);

File: scope.py
x = 10  # Global variable

def outer():
    x = 20  # Enclosing scope
    def inner():
        x = 30  # Local scope
        print("Inner x:", x)
    inner()
    print("Outer x:", x)

if __name__ == "__main__":
    outer()
    print("Global x:", x)
